map/transit/transit_reader.cpp

```diff
@@ -1,108 +1,57 @@
-    ReadHeader();
-  return IsValid();
-}
-
-bool TransitReader::IsValid() const
-{
-  return m_src != nullptr;
-}
-
-unique_ptr<TransitReader::TransitReaderSource> TransitReader::GetReader(MwmSet::MwmId const & mwmId)
-{
-  MwmSet::MwmHandle handle = m_index.GetMwmHandleById(mwmId);
-  if (!handle.IsAlive())
-  {
-    LOG(LWARNING, ("Can't get mwm handle for", mwmId));
-    return unique_ptr<TransitReaderSource>();
-  }
-  MwmValue const & mwmValue = *handle.GetValue<MwmValue>();
-  if (!mwmValue.m_cont.IsExist(TRANSIT_FILE_TAG))
-    return unique_ptr<TransitReaderSource>();
-  FilesContainerR::TReader reader = mwmValue.m_cont.GetReader(TRANSIT_FILE_TAG);
-  return my::make_unique<TransitReaderSource>(reader);
-}
-
-void TransitReader::ReadHeader()
-{
-  try
-  {
-    transit::FixedSizeDeserializer<ReaderSource<FilesContainerR::TReader>> fixedSizeDeserializer(*m_src.get());
-    fixedSizeDeserializer(m_header);
-  }
-  catch (Reader::OpenException const & e)
-  {
-    LOG(LERROR, ("Error while reading transit header.", e.Msg()));
-    throw;
-  }
-}
-
-void TransitReader::ReadStops(vector<transit::Stop> & stops)
-{
-  ReadTable(m_header.m_stopsOffset, stops);
-}
-
-void TransitReader::ReadShapes(vector<transit::Shape> & shapes)
-{
-  ReadTable(m_header.m_shapesOffset, shapes);
-}
-
-void TransitReader::ReadTransfers(vector<transit::Transfer> & transfers)
-{
-  ReadTable(m_header.m_transfersOffset, transfers);
-}
-
-void TransitReader::ReadLines(vector<transit::Line> & lines)
-{
-  ReadTable(m_header.m_linesOffset, lines);
-}
-
-void TransitReader::ReadNetworks(vector<transit::Network> & networks)
-{
-  ReadTable(m_header.m_networksOffset, networks);
-}
-
-bool ReadTransitTask::Init(uint64_t id, MwmSet::MwmId const & mwmId, unique_ptr<TransitDisplayInfo> && transitInfo)
+// ReadTransitTask --------------------------------------------------------------------------------
+void ReadTransitTask::Init(uint64_t id, MwmSet::MwmId const & mwmId, unique_ptr<TransitDisplayInfo> && transitInfo)
 {
   m_id = id;
   m_mwmId = mwmId;
   if (transitInfo == nullptr)
   {
     m_loadSubset = false;
-    m_transitInfo = make_unique<TransitDisplayInfo>();
+    m_transitInfo = my::make_unique<TransitDisplayInfo>();
   }
   else
   {
     m_loadSubset = true;
     m_transitInfo = move(transitInfo);
   }
-  return m_transitReader.Init(mwmId);
 }
 
 void ReadTransitTask::Do()
 {
-  if (!m_transitReader.IsValid())
+  MwmSet::MwmHandle handle = m_index.GetMwmHandleById(m_mwmId);
+  if (!handle.IsAlive())
+  {
+    LOG(LWARNING, ("Can't get mwm handle for", m_mwmId));
+    return;
+  }
+  MwmValue const & mwmValue = *handle.GetValue<MwmValue>();
+  if (!mwmValue.m_cont.IsExist(TRANSIT_FILE_TAG))
     return;
 
-  vector<transit::Stop> stops;
-  m_transitReader.ReadStops(stops);
-  FillItemsByIdMap(stops, m_transitInfo->m_stops);
-  stops.clear();
+  auto reader = my::make_unique<FilesContainerR::TReader>(mwmValue.m_cont.GetReader(TRANSIT_FILE_TAG));
```

Comment:
Why do we need unique ptr here? Why not just
```c++
auto reader = mwmValue.m_cont.GetReader(...)`
```?

Reply:
No. Not at all. Done.